### PR TITLE
New persistence client bean for DB clients management

### PR DESCRIPTION
**Pull request by wxing1292**

### CODE REVIEW

File: common/persistence/client/bean.go [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+		historyManager          persistence.HistoryManager
+		executionManagerFactory persistence.ExecutionManagerFactory
+
+		sync.RWMutex
+		shardIDToExecutionManager map[int]persistence.ExecutionManager
+	}
+)
+
+// NewBeanByFactory crate a new store bean using factory
+func NewBeanByFactory(
```
Comment (longquanzheng) [inline common/persistence/client/bean.go]: How about NewBeanFromFactory?

File: common/persistence/client/factory.go [hunk 2]
```diff
@@ -44,8 +44,8 @@ type (
 		NewTaskManager() (p.TaskManager, error)
 		// NewShardManager returns a new shard manager
 		NewShardManager() (p.ShardManager, error)
-		// NewHistoryV2Manager returns a new historyV2 manager
-		NewHistoryV2Manager() (p.HistoryManager, error)
+		// NewHistoryManager returns a new historyV2 manager
```
Comment (longquanzheng) [inline common/persistence/client/factory.go]: could you also fix the comment

File: common/persistence/client/bean.go [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+import (
+	"sync"
+
+	"github.com/uber/cadence/common/persistence"
+)
+
+type (
+	// Bean in an collection of persistence manager
+	Bean interface {
```
Comment (longquanzheng) [inline common/persistence/client/bean.go]: Is it being used anywhere in this PR? or just for future?

Comment (wxing1292) [reply common/persistence/client/bean.go]: i have a really large commit doing service startup logic refactoring, this is part of that

File: common/persistence/client/factory.go (hunk 2 above)
Comment (wxing1292) [reply common/persistence/client/factory.go]: sure

File: common/persistence/client/bean.go (hunk 1 above)
Comment (wxing1292) [reply common/persistence/client/bean.go]: sure

File: common/persistence/client/factory.go [hunk 4]
```diff
@@ -18,7 +18,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package persistence
+package client
```
Comment (yycptt) [inline common/persistence/client/factory.go]: Can we rename the `New()` function in this file to maybe `NewFactory()`? I think `client.New()` returning a persistence factory a little bit counter-intuitive.

Comment (wxing1292) [reply common/persistence/client/factory.go]: done

**Pull request merged**